fl4health/model_bases/pca.py

```diff
@@ -35,32 +63,40 @@ def forward(self, X: Tensor, center_data: bool) -> Tuple[Tensor, Tensor]:
 
         Args:
             X (Tensor): Data matrix.
-            center_data (bool): If true, then the data mean will be subtracted from all data points prior to
-            performing PCA.
+            center_data (bool): If true, then the data mean will be subtracted
+            from all data points prior to performing PCA. If center_data is false,
+            it is expected that the data has already been centered and an exception
```

Comment:
I don't think we're throwing an exception. So I assume the underlying method calls will throw one?

Reply:
Yes you are right. I've updated `prepare_data_forward()` to check whether the data already has zero mean and throw an exception accordingly.